### ISSUE TITLE
Switch to v2 breaks app behvior

### ISSUE THREAD

**Issue by dartmann** [nextcloud/android-library#768]
Hi,

I decided to migrate to the new version v2 of the nextcloud android-library (before I used v1.0.40). After this I changed all classes, e. g. from `UploadRemoteFileOperation` to `UploadFileRemoteOperation` and thought I was ready to go.

Now I can not read/upload files any more. When I execute an UploadFileRemoteOperation the logs show me this:
`OwnCloudClient #1: REQUEST PUT /remote.php/dav/files/null//upload_folder/image.jpg` followed by: `Local file does not exist`. The same applies when executing ReadFileRemoteOperation. I noticed the remote path is not correct as it contains this `null/`. This should be the user ID, right? How can I correct this? Even if the username is set correctly, we have an additional slash here...

I initialize the client via:
```java
final OwnCloudClient client = OwnCloudClientFactory.createOwnCloudClient(
                Uri.parse(context.getString(R.string.config_nextcloud_server_base_url)),
                context, true);
        client.setCredentials(OwnCloudCredentialsFactory
                .newBasicCredentials(context.getString(R.string.config_nextcloud_user_name),
                        context.getString(R.string.config_nextcloud_api_token)));
```

So after fiddling around with the new library version I am pretty sure that my problem is because of the different approach of `NextcloudClient` (amongst others it uses the user ID instead of username and password. I did not work with this before and additionally was not able to find any examples. Furthermore the documentation lacks this completely and is still based on the old version of the lib.).

To sum it up: it would be great if you could update your documentation on how to use the lib and provide examples for the different use cases (CRUD-Ops) on basis of `NextcloudClient` respectively latest version of the lib.

Atm I reverted back to latest v1 (1.7.0) to get my app working again. Btw: it uses a different webdav path: `/remote.php/webdav/upload_folder/`.

Thank you very much for your support. I am top motivated to help the project here, if I can, of course :)
Kind regards, David

**Comment by MonkeySon** [comment]
I could observe a similar behavior and found out the following:

I use a simple `ReadFolderRemoteOperation()` to read the root directory of my nextcloud.

Up to version **2.7.1** it works, and Logcat outputting:

```
D/OwnCloudClient #0: REQUEST PROPFIND /remote.php/webdav/
...
I/ReadFolderRemoteOperation: Synchronized /: Operation finished with HTTP status code 207 (success)
```

Anything from **2.8.0 and above** does not work:
```
D/OwnCloudClient #0: REQUEST PROPFIND /remote.php/dav/files/null//
...
E/ReadFolderRemoteOperation: Synchronized /: Local file does not exist
```

**Comment by tobiasKaminsky** [comment]
> /remote.php/dav/files/null//

This means that you do not have set userId.
You can set it directly in OwncloudClient via "setUserId" or use a fitting function of OwnCloudClientFactory.
